### PR TITLE
Pull Request for Issue1507: Update REIXS/IDEAS buildStandardExportOption 

### PR DESCRIPTION
**Pull request by epengr**
updated the buildStandardExporter strings for REIXS and IDEAS
Added a couple controls to REIXSBeamline
Exposed additional control in IDEASBeamline
Tweaked all ScanInitialConditions to support new buildStandardExporter configs
Added a new keywork replacers to handle Enums and Enum replacer methods in scanConfigurations

### CODE REVIEW

File: source/acquaman/IDEAS/IDEASXASScanActionController.cpp [hunk 1]
```diff
@@ -217,18 +219,16 @@ AMAction3* IDEASXASScanActionController::createInitializationActions()
...
 	positions.remove(positions.indexOf("XRF1E Real Time"));
-
-	if(!configuration_->fluorescenceDetector().testFlag(IDEAS::NoXRF)){
-
-		positions.remove(positions.indexOf("XRF1E Peaking Time"));
-		positions.remove(positions.indexOf("XRF1E Trigger Level"));
-		positions.remove(positions.indexOf("XRF1E Baseline Threshold"));
-		positions.remove(positions.indexOf("XRF1E Preamp Gain"));
-	}
+	positions.remove(positions.indexOf("13-element Ge dwell time"));
```
Comment (dretrex) [inline source/acquaman/IDEAS/IDEASXASScanActionController.cpp]: Does this not need to be inside an `if statement`?

File: source/acquaman/REIXS/REIXSXASScanConfiguration.cpp [hunk 2]
```diff
@@ -104,6 +104,98 @@ REIXSXASScanConfiguration::REIXSXASScanConfiguration(const REIXSXASScanConfigura
...
+			return "Linear Inclined";
+			break;
+		}
+	}
+	else if(enumName == "monoMirrorSelector"){
+		switch(enumValue){
+		case 0:
+			return "Nickle";
+			break;
+		case 1:
```
Comment (dretrex) [inline source/acquaman/REIXS/REIXSXASScanConfiguration.cpp]: You don't need `break;` statements if you are using return inside the `case` statements.

File: source/acquaman/REIXS/REIXSXESScanConfiguration.cpp [hunk 3]
```diff
@@ -93,6 +93,100 @@ REIXSXESScanConfiguration::REIXSXESScanConfiguration(const REIXSXESScanConfigura
...
+		return true;
+	return false;
+}
+
+QString REIXSXESScanConfiguration::enumConvert(const QString &enumName, int enumValue) const{
+	if(enumName == "spectrometerGrating"){
+		switch(enumValue){
+		case 0:
+			return "LEG";
+			break;
```
Comment (dretrex) [inline source/acquaman/REIXS/REIXSXESScanConfiguration.cpp]: Same here.

File: source/beamline/REIXS/REIXSBeamline.cpp [hunk 4]
```diff
@@ -97,6 +97,9 @@ REIXSBeamline::REIXSBeamline() :
 	tfyDetector_ = new CLSBasicScalerChannelDetector("TFY", "TFY", scaler_, 4, this);
 	pfyDetector_ = new CLSBasicScalerChannelDetector("PFY", "PFY", scaler_, 3, this);
 
+	I0Current_ = new AMReadOnlyPVControl("I0Current","BL1610-ID-2:mcs16:fbk", this, "I0 Current");
```
Comment (dretrex) [inline source/beamline/REIXS/REIXSBeamline.cpp]: Not the biggest deal, but member variables should start with lower case (i.e.: `i0Current_` and `teyCurren_`).

**Comment by dretrex** [comment]
Overall, looks good.
Time Estimate: 10 minutes

File: source/acquaman/IDEAS/IDEASXASScanActionController.cpp (hunk 1 above)
Comment (epengr) [inline source/acquaman/IDEAS/IDEASXASScanActionController.cpp]: Not really.  It just means that those settings are being recorded for scans that aren't using that detector, which isn't an issue.

File: source/acquaman/REIXS/REIXSXASScanConfiguration.cpp (hunk 2 above)
Comment (epengr) [inline source/acquaman/REIXS/REIXSXASScanConfiguration.cpp]: True.   I really should start looking at the code I steal from others more carefully:


```

QString SGMXASScanConfiguration::enumConvert(const QString &enumName, int enumValue) const{
if(enumName == "grating"){
switch(enumValue){
case 0:
return "Low";
break;
case 1:
return "Medium";
break;
case 2:
return "High";
break;
}
}...
```

**Pull request merged**